framework/src/interfacekernels/InterfaceReaction.C

```diff
@@ -0,0 +1,89 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceReaction.h"
+
+registerMooseObject("MooseApp", InterfaceReaction);
+
+template <>
+InputParameters
+validParams<InterfaceReaction>()
+{
+  InputParameters params = validParams<InterfaceKernel>();
+  params.addParam<MaterialPropertyName>("D", "D", "The diffusion coefficient.");
+  params.addParam<MaterialPropertyName>(
+      "D_neighbor", "D_neighbor", "The neighboring diffusion coefficient.");
+  params.addRequiredParam<Real>("kf", "Forward reaction rate coefficient.");
+  params.addRequiredParam<Real>("kb", "Backward reaction rate coefficient.");
+  params.addClassDescription("Implements a reaction to establish ReactionRate=k_f*u-k_b*v "
+                             "at interface.");
+  return params;
+}
+
+InterfaceReaction::InterfaceReaction(const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _kf(getParam<Real>("kf")),
+    _kb(getParam<Real>("kb")),
+    _D(getMaterialProperty<Real>("D")),
+    _D_neighbor(getNeighborMaterialProperty<Real>("D_neighbor"))
+{
+}
+
+Real
+InterfaceReaction::computeQpResidual(Moose::DGResidualType type)
+{
+  Real r = 0;
+
+  switch (type)
+  {
+
+      // Move all the terms to the LHS to get residual, for master domain
+      // Residual = kf*u - kb*v = kf*u - kb*v
+      // Weak form for master domain is: (test, kf*u - kb*v)
+
```

Comment:
Remove this space and then make sure that comment indentation matches that of the code below it